### PR TITLE
Serialize enriched metadata - Part 2

### PR DESCRIPTION
**Pull request by thomasabraham**
# Introduced dedicated `NakadiRecordResource` class

> Zalando ticket : team-aruha/issues/1106

## Review
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class NakadiRecordResource implements Resource<NakadiRecord> {
+    private final NakadiRecord nakadiRecord;
+
+    public NakadiRecordResource(final NakadiRecord nakadiRecord) {
+        this.nakadiRecord = nakadiRecord;
+    }
+
+    @Override
+    public String getName() {
+        return nakadiRecord.getMetadata().getEid();
```
Comment (a1exsh) [inline core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java:21]: Now I recall I wanted to ask about this line :p

Comment (adyach) [reply core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java:21]: that's copy from [batch item](https://github.com/zalando/nakadi/blob/9f52df9c63136af9255f18e7725c553c4ebf40a5/core-common/src/main/java/org/zalando/nakadi/domain/BatchItem.java#L156).

Comment (a1exsh) [reply core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java:21]: Can I unsee that, please?.. :see_no_evil: 

That still doesn't answer what's going on there.  Why `eid` makes sense as a "name" of resource in context of Event-based auth?

**Comment by a1exsh** [comment]
:+1:

File: core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java (hunk 1 above)
Comment (thomasabraham) [reply core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecordResource.java:21]: AFAIK, we are (mis)using the Authorization concepts of spring security, which works fine for event-types when we originally started using it. An event-type is identified by its name, and the `getName` function in the interface is returns the event-type's name in Nakadi's `ResourceImpl` when used for an event-type; and subscription-id in the case of subscriptions (which are introduced to Nakadi later). See `asResource()` method is `EventType` and `Subscription`. 

Then much later we implemented per event authorization. In order to re-use the existing concepts, plugins and libraries, now we have to represent an event as a `Resource`, just like how we do with `EventType` and `Subscription`.
So the `BatchItem` implements `Resource` in which `getName()` returns `eid` of the event. And naturally this gets copied to the Avro implementation of `BatchItem`, which is `NakadiRecord`.

**Comment by thomasabraham** [comment]
:+1:

**Pull request merged**